### PR TITLE
Add support for changing passwords

### PR DESCRIPTION
**Pull request by not-an-aardvark**
This bit is obviously pretty important security-wise, so hopefully I didn't mess anything up.

Builds off of #76, so look at that first.

### CODE REVIEW

**Comment by YaManicKill** [comment]
Alrighty, can you rebase this so I can see just these changes now I've merged #76. Thanks :-)

**Comment by not-an-aardvark** [comment]
Rebased

File: test/controller/auth.js [hunk 1]
```diff
@@ -153,4 +153,108 @@ describe('AuthController', function() {
...
+  describe('changing a password', () => {
+    let passAgent, passAgent2, passAgent3, username;
+    beforeEach(async () => {
+      username = `USERNAME_${require('crypto').randomBytes(4).toString('hex')}`;
+      passAgent = supertest.agent(sails.hooks.http.app);
+      passAgent2 = supertest.agent(sails.hooks.http.app);
+      passAgent3 = supertest.agent(sails.hooks.http.app);
+      const res = await passAgent.post('/auth/local/register').send({
+        name: username,
+        password: 'Correct Horse Battery Staple',
```
Comment (YaManicKill) [inline test/controller/auth.js]: :+1: good one

File: api/controllers/UserController.js [hunk 2]
```diff
@@ -75,6 +75,40 @@ module.exports = _.mapValues({
...
+      return res.forbidden('Incorrect password');
+    }
+    // Otherwise, attempt to create a new Passport for the user with the new password.
+    // This is done *before* deleting the old Passport; otherwise, the user will be locked out of their
+    // account if there's an error creating the new Passport.
+    const newPassport = await Passport.create({
+      protocol: 'local',
+      user: req.user.name,
+      password: params.newPassword,
+      accessToken: require('crypto').randomBytes(48).toString('base64')
```
Comment (YaManicKill) [inline api/controllers/UserController.js]: Yeah, I think this is the right way to do things, it looks the same as how they set up the registration, so, looks good to me.

**Comment by YaManicKill** [comment]
Yeah, it looks good. We can always find an expert on these things before we go live if we feel the need, but you are following the way that sails-auth set up the registration, so looks good.

**Pull request merged**